src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseTestCaseService.java

```diff
@@ -148,13 +170,21 @@ public Result updateResultFromTestCases(Result result, ProgrammingExercise exerc
             // Recalculate the achieved score by including the test cases individual weight.
             if (successfulTestCases.size() > 0) {
                 long successfulTestScore = successfulTestCases.stream().map(ProgrammingExerciseTestCase::getWeight).mapToLong(w -> w).sum();
-                long maxTestScore = testCases.stream().map(ProgrammingExerciseTestCase::getWeight).mapToLong(w -> w).sum();
+                long maxTestScore = testCasesForCurrentDate.stream().map(ProgrammingExerciseTestCase::getWeight).mapToLong(w -> w).sum();
```

Comment:
Out of interest: Why is the score a of type long?

Reply:
I think this is the recommendation from jhipster